src/vm/eventpipefile.h

```diff
@@ -170,6 +173,10 @@ class EventPipeFile final : public FastSerializableObject
 
     Volatile<LONG> m_metadataIdCounter;
 
+    // Indicates whether the file has been initialized,
+    // e.g., has the header been sent to the stream?
+    Volatile<BOOL> m_isInitialized;
```

Comment:
A bit of a high-level question here.... `m_isInitialized` seems to be used only for asserting. Is that intentional? If so, do you want to put this under #ifdef to make it debug only thing?